execute/exectypes/outcome.go

```diff
@@ -104,10 +104,7 @@ func NewSortedOutcome(
 	sort.Slice(
 		pendingCommitsCP,
 		func(i, j int) bool {
-			if pendingCommitsCP[i].SourceChain != pendingCommitsCP[j].SourceChain {
-				return pendingCommitsCP[i].SourceChain < pendingCommitsCP[j].SourceChain
-			}
-			return pendingCommitsCP[i].SequenceNumberRange.Start() < pendingCommitsCP[j].SequenceNumberRange.Start()
+			return pendingCommitsCP[i].Timestamp.Before(pendingCommitsCP[j].Timestamp)
```

Comment:
This is fine for now, thinking how can we make this much clearer and less cluttery in the future (and also less prone to bugs). The `Test` that tests the timestamp ordering is a good start.